omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/VisitConfigurationController2_0Test.java

```diff
@@ -0,0 +1,110 @@
+	}
+
+	@Test
+	public void shouldGetCurrentConfiguration() throws Exception {
+		// set initial configuration
+
+		// enable visits
+		GlobalProperty enableVisitsGp = new GlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_ENABLE_VISITS, "true");
+		administrationService.saveGlobalProperty(enableVisitsGp);
+
+		// set encounter visit handler
+		GlobalProperty visitEncounterHandlerGp = new GlobalProperty(OpenmrsConstants.GP_VISIT_ASSIGNMENT_HANDLER,
+				"org.openmrs.api.handler.NoVisitAssignmentHandler");
+		administrationService.saveGlobalProperty(visitEncounterHandlerGp);
+
+		// set visit types to auto close
+		VisitType testVisitType = visitService.getVisitTypeByUuid(RestTestConstants1_9.VISIT_TYPE_UUID);
+		GlobalProperty visitTypesToAutoCloseGp = new GlobalProperty(OpenmrsConstants.GP_VISIT_TYPES_TO_AUTO_CLOSE,
+				testVisitType.getName());
+		administrationService.saveGlobalProperty(visitTypesToAutoCloseGp);
+
+		// make GET call
+		MockHttpServletRequest req = request(RequestMethod.GET, "visitconfiguration");
+		SimpleObject result = deserialize(handle(req));
+
+		// assert response
+		assertNotNull(result);
+		assertEquals(true, PropertyUtils.getProperty(result, "enableVisits"));
+		assertEquals("org.openmrs.api.handler.NoVisitAssignmentHandler",
+				PropertyUtils.getProperty(result, "encounterVisitsAssignmentHandler"));
+
+		List<Object> visitTypesToAutoClose = (List<Object>) PropertyUtils.getProperty(result, "visitTypesToAutoClose");
+		assertEquals(1, visitTypesToAutoClose.size());
+	}
+
+	@Test
+	public void shouldUpdateCurrentConfiguration() throws Exception {
+		// assert initial configuration
+		MockHttpServletRequest req = request(RequestMethod.GET, "visitconfiguration");
+		SimpleObject result = deserialize(handle(req));
+
+		assertNotNull(result);
+		assertEquals(false, PropertyUtils.getProperty(result, "enableVisits"));
+		assertNull(PropertyUtils.getProperty(result, "encounterVisitsAssignmentHandler"));
+		List<Object> visitTypesToAutoClose = (List<Object>) PropertyUtils.getProperty(result, "visitTypesToAutoClose");
+		assertEquals(0, visitTypesToAutoClose.size());
+
+		// update configuration
+		String json =
+				"{\"enableVisits\": true,\"encounterVisitsAssignmentHandler\": \"org.openmrs.api.handler.NoVisitAssignmentHandler\",\"startAutoCloseVisitsTask\": true,\"visitTypesToAutoClose\":[{\"uuid\":\""
+						+ RestTestConstants1_9.VISIT_TYPE_UUID + "\"}]}";
+		handle(newPostRequest("visitconfiguration", json));
+
+		// make GET call
+		result = deserialize(handle(req));
+
+		// assert response
+		assertNotNull(result);
+		assertEquals(true, PropertyUtils.getProperty(result, "enableVisits"));
+		assertEquals("org.openmrs.api.handler.NoVisitAssignmentHandler",
+				PropertyUtils.getProperty(result, "encounterVisitsAssignmentHandler"));
+
+		visitTypesToAutoClose = (List<Object>) PropertyUtils.getProperty(result, "visitTypesToAutoClose");
+		assertEquals(1, visitTypesToAutoClose.size());
```

Comment:
Can you also check the value?

Reply:
Resolved here: https://github.com/openmrs/openmrs-module-webservices.rest/pull/489/commits/b5a0413a61da151b98e80e62325bfa3582a73820